metering-billing/lambdas/src/main/java/com/amazon/aws/partners/saasfactory/metering/onboarding/BillingIntegration.java

```diff
@@ -366,25 +388,47 @@ private void cancelSubscriptionInStripe(String tenantId) throws StripeException
             final SubscriptionListParams subscriptionListParams = SubscriptionListParams.builder().setCustomer(customer.getId()).build();
             final SubscriptionCollection collection = Subscription.list(subscriptionListParams);
             for (final Subscription subscription : collection.getData()) {
+                //NOTE: Do queued metered events need to sent to Stripe before canceling the subscription?
+
                 //attempt to cancel the subscription
-                LOGGER.info("disableTenantInStripe: Cancel Subscription: {} ", subscription.getId());
+                LOGGER.info("cancelSubscriptionInStripe: Cancel Subscription: {} ", subscription.getId());
                 Subscription canceledSubscription = subscription.cancel();
             }
         } catch (StripeException se) {
             //this means cancel failed
-            LOGGER.error("Error canceling subscription in Stripe for customer: {}", customerId);
+            LOGGER.error("Error canceling subscription in Stripe for customer: {}", customerName);
             LOGGER.error(Utils.getFullStackTrace(se));
         }
     }
 
+    private Customer getStripeCustomer(final String customerId) throws StripeException {
+        //Only attempt to disable tenant if customer exists
+        Map<String, Customer> custIdMap = new LinkedHashMap<>();
+        boolean hasMore = false;
+        String startingAfter = null;
+        LOGGER.info("Find Stripe customer with name: {} ", customerId);
+        do {
+            CustomerListParams customerListParams = CustomerListParams.builder().setLimit(100l).setStartingAfter(startingAfter).build();
+            CustomerCollection customerCollection = Customer.list(customerListParams);
+            hasMore = customerCollection.getHasMore();
+            for (Customer customer : customerCollection.getData()) {
+                if (customerId.equals(customer.getName())) {
+                    return customer;
+                }
+                startingAfter = customer.getId();
+            }
+        } while (hasMore);
+        return null;
+    }
+
 
     /*
-    Creates a customer and a subscription in Stripe
+    Creates a subscription in Stripe
     The Subscription will have one item for the monthly plan and additional items for metered products
     For the metered products, a message is sent to Event Bus to add that record to the Dynanamo table mapping
      the Tenant internal product code to the Stripe Subscription id which is unique for each tenant and product.
      */
-    private void createStripeSubscription(Customer customer, SubscriptionPlan plan, String fullTenantId) throws StripeException{
+    private void createStripeSubscription(Customer customer, SubscriptionPlan plan, String fullTenantId, boolean existingCustomer) throws StripeException{
```

Comment:
We're now passing in `existingCustomer` but it doesn't appear you're using it.

Reply:
Updated to remove unused variable